### PR TITLE
KDS: kolibri-tools jest.conf & extract all KeenUI to KDS & KButtonGroup margin update

### PR DESCRIPTION
**Pull request by nucleogenesis**
### Summary

**This PR has been hijacked by other issues**

- kolibri-tools jest.conf updated to use @babel - includes aphrodite change that "makes it work" yay!
- jest snapshots updated en masse
- This branch currently points to a specific KDS branch on my fork for now... will change that soon.

Generally speaking - we ran into a large slew of errors around how Jest handled KDS. Now that is fixed with a huge thanks to @jonboiser - huge thanks to @jonboiser and @rtibbles and @indirectlylit for the help finding a resolution to what has held this up for a month.

**Below this is the original PR's description**

KButtonGroup will only apply margins internally - so the top/bottom margins will be removed and the onus is on the implementer to determine what, if any, vertical margins a KButtonGroup needs.

Note that KButtonGroup is an `inline` element - so y-axis margins applied will not affect the wrapping `<span>` element's spacing.

Such margins must be applied to the children directly. This means any `KButton KRouterLink KExternalLink` or `<a> <button>` children must have vertical margins applied to them directly.

For speed I've just applied these as inline styles directly on all child components - given general approval of this PR I'll go through a few components and extract the repeated margins to a class to be shared by the child buttons.

The reasoning for this is that, in general, we do not want Design System components to bring external margins with them. In Kolibri, the 8px vertical margins were universally used, but in other applications this may not always be the case.

*BONUS COMMIT:*

- The "View All" buttons on Coach > Class Summary were misaligned due to some recent change that I made. Updated the `right` CSS property to realign them properly.

### Reviewer guidance

Follow the general "how to test" here: https://github.com/learningequality/kolibri-design-system/pull/16

Check the files affected and ensure that the grouped buttons align properly vertically and horizontally.

Also check out Coach > Class Home's "View all" buttons in the left-sided blocks.

### References

https://github.com/learningequality/kolibri-design-system/pull/16

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/views/ExamPage/index.vue [hunk 1]
```diff
@@ -206,11 +205,11 @@
       bottomBarLayoutDirection() {
         // Allows contents to be displayed visually in reverse-order,
         // but semantically in correct order.
-        return this.isRtl ? 'ltr' : 'rtl';
+        return this.isRtl ? 'rtl' : 'ltr';
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/views/ExamPage/index.vue]: is this correct? Let's discuss

**Comment by nucleogenesis** [comment]
@indirectlylit seems like travis will be mad no matter what until a new release is cut and package.json points to it.

**Comment by nucleogenesis** [comment]
@indirectlylit I've had issues getting the same errors locally as are showing in the `node10.x` environment. However, I did realize that the `kolibri-design-system` files were being ignored by the Jest config which I changed in the last commit.

I'm not having any luck trying to unwind this mess of dependencies. Going to pivot for sanity's sake - will return to it later or early tomorrow.


Also:

The errors in the Travis `lint` environment will be resolved whenever we point the package.json to a new release of KDS. It's just looking for a new file that doesn't exist in the latest release yet.

**Comment by indirectlylit** [comment]
Okay, ping me on slack if you'd like me to take a swing at it

**Comment by indirectlylit** [comment]
we can also merge this in and address in a subsequent PR to the design system

**Comment by nucleogenesis** [comment]
Closed in favor of this PR:

https://github.com/learningequality/kolibri/pull/6898

**Pull request closed without merging**